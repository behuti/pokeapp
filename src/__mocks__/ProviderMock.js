//This mock should homologate the index with the provider
import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import rootReducer from "./../redux/rootReducer";

const store = createStore(rootReducer);

const ProviderMock = (props) => (
    <Provider store={store}>{props.children}</Provider>
);

export default ProviderMock;
