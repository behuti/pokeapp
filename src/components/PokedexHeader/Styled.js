import styled from "styled-components";

export const Header = styled.div`
    position: relative;
    height: 130px;
    margin-bottom: 40px;
    max-width: 100%;
    overflow: hidden;

    .pokemon-search__blueIndicator {
        position: absolute;
        display: block;
        top: 40px;
        left: 40px;
        width: 25px;
        height: 25px;
        box-shadow: ${({ theme }) =>
            theme.pixelatedBorderRadius.step4(4, "#323232", "#46B5F9")};
        background-color: #46b5f9;

        &::after {
            content: "";
            position: absolute;
            top: 0px;
            left: 0px;
            display: block;
            width: 10px;
            height: 10px;
            background-color: #fff;
            box-shadow: ${({ theme }) =>
                theme.pixelatedBorderRadius.step1(4, "#fff")};
            z-index: 1;
        }
    }

    .pokemon-search__redIndicator {
        display: block;
        position: absolute;
        top: 45px;
        left: 115px;
        width: 11px;
        height: 11px;
        background-color: red;
        box-shadow: ${({ theme }) =>
            theme.pixelatedBorderRadius.step2(3, "#323232", "red")};
    }

    .pokemon-search__yellowIndicator {
        display: block;
        position: absolute;
        top: 45px;
        left: 145px;
        width: 11px;
        height: 11px;
        background-color: #ffa608;
        box-shadow: ${({ theme }) =>
            theme.pixelatedBorderRadius.step2(3, "#323232", "#ffa608")};
    }

    .pokemon-search__greenIndicator {
        display: block;
        position: absolute;
        top: 45px;
        left: 175px;
        width: 11px;
        height: 11px;
        background-color: #39b348;
        box-shadow: ${({ theme }) =>
            theme.pixelatedBorderRadius.step2(3, "#323232", "#39b348")};
    }

    .pokemon-search__separator {
        position: absolute;
        left: 0;
        bottom: 0;
    }
`;
