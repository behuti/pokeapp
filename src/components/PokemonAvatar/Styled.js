import styled from "styled-components";

export const StyledPokemonAvatar = styled.div`
    text-align: center;
    background-color: white;
    max-width: 400px;
    margin: 0 auto;
    padding: 30px 30px 60px; 
    box-shadow: 0 4px black, 0 -4px black, 4px 0 black, -4px 0 black, 4px 9px #891515, 9px 4px #891515;
`;

export const StyledPokemonScreen = styled.div`
    position: relative;
    min-height: 300px;
    box-shadow: 0 4px black, 0 -4px black, 4px 0 black, -4px 0 black, 0 9px #D2E5E3, 9px 0 #D2E5E3;
    background-color: #132553;
    ${({ theme, displayOn, type }) =>
        displayOn &&
        `
        background: ${ theme.pokemonGradient(theme.colors.types[type]) };
    `}

    .pokemon-avatar__picture {
        width: 150px;
        margin-top: 25px;

        &--error {
            width: 150px;
            display: block;
            position: absolute;
            top: 30px;
            left: 50%;
            transform: translateX(-50%);

            & + .pokemon-avatar__picture {
                margin-top: 60px;
            }

            &:after {
                content: 'ZZzzZZzz...';
                color: white;
            }
        }
    }
    .pokemon-screen__indicator {
        position: absolute;
        bottom: -40px;
        width: 15px;
        height: 15px;
        background-color: #F0060E;
        box-shadow: 0 4px #F0060E, 0 -4px #F0060E, 4px 0 #F0060E, -4px 0 #F0060E, 0 9px black, 9px 0 black, 0 -9px black, -9px 0 black, 5px 5px black, -5px -5px black, -5px 5px black, 5px -5px black ;
    }
    .pokemon-screen__speaker {
        position: absolute;
        bottom: -50px;
        width: 40px;
        right: 0;

        .line {
            height: 4px;
            width: 100%;
            margin-bottom: 4px;
            background-color: black;
        }
    }
`;

export const StyledPokemonName = styled.h3`
    position: absolute;
    bottom: 0;
    width: 100%;
    text-transform: uppercase;
    letter-spacing: 2.5px;
    color: white;
    background-color: #1c3336;
    padding: 20px 0;
    margin-bottom: 0;
`;

export const StyledPokemonType = styled.span`
    display: inline-block;
    padding: 5px 4px;
    font-size: 10px;
    background-color: ${(props) =>
        props.type && props.theme.colors.types[props.type]};
    color: white;
`;
