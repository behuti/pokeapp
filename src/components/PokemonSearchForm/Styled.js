import styled from 'styled-components';

export const StyledForm = styled.div`
    text-align: center;
    padding: 140px 0;
    position: relative;
`;

export const StyledButton = styled.button`
    color: red;
`;