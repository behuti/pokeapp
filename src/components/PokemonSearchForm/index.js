import React, { useState } from "react";
import { connect } from "react-redux";
import { fetchPokemon } from "../../redux/pokemon/pokemonActions";
import Theme from "../../styles/theme";

import { StyledForm, StyledButton } from "./Styled";

const PokemonSearchForm = ({fetchPokemon}) => {
    const [pokeName, setPokeName] = useState("");

    return (
        <Theme>
            <StyledForm className="pokemon-search">
                <h1>What Pokemon are you looking for?</h1>
                <input
                    type="text"
                    value={pokeName}
                    onChange={(e) => setPokeName(e.target.value)}
                />
                <StyledButton onClick={() => {
                  console.log('hola')
                  fetchPokemon(pokeName)
                  }}>
                    Search pokemon
                </StyledButton>
            </StyledForm>
        </Theme>
    );
};

const mapDispatchToProps = (dispatch) => ({
    fetchPokemon: (pokemonName) => dispatch(fetchPokemon(pokemonName)),
});

export default connect(null, mapDispatchToProps)(PokemonSearchForm);
