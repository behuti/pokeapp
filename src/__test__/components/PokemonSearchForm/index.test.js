import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProviderMock from "../../../__mocks__/ProviderMock";
import PokemonSearchForm from "../../../components/PokemonSearchForm";
import { SearchButton } from "../../../components/PokemonSearchForm/Styled";

describe("<PokemonSearchForm />", () => {
    const wrapper = mount(
        <ProviderMock>
            <PokemonSearchForm />
        </ProviderMock>
    );
    test("Should render", () => {
        expect(wrapper.length).toEqual(1); //Check that the General component is rendering
    });
    test("should render the title", () => {
        const title = wrapper.find("h1");
        expect(title.text()).toEqual("What Pokemon are you looking for?");
    });
    test("should render the form", () => {
        const form = wrapper.find("form");
        expect(form.length).toEqual(1);
    });

    test("should simulate pokemon Name field and clear it", () => {
        let textField = wrapper.find(".pokemon-search__input");
        let clearButton = wrapper.find("#clearButton");
        textField.simulate("change", {
            target: { value: "25" },
        });
        textField = wrapper.find("#pokemonName"); //This is because the input changes while the wrapper updates
        expect(textField.props().value).toEqual("25");
        clearButton = wrapper.find("#clearButton").first();
        //Now clear the input
        clearButton.simulate("click");
        textField = wrapper.find("#pokemonName"); //This is because the input changes while the wrapper updates
        expect(textField.props().value).toEqual("");
    });
});

describe("<SearchButton/>", () => {
    const onSubmitHandler = jest.fn();
    const wrapper = mount(
        <ProviderMock>
            <SearchButton onClick={onSubmitHandler} />
        </ProviderMock>
    );
    test("should fire onClick", () => {
        wrapper.simulate("click");
        expect(onSubmitHandler).toBeCalledTimes(1);
    });
});

describe("Form Snapshot", () => {
    test("should match snapshot", () => {
        const form = create(
            <ProviderMock>
                <PokemonSearchForm />
            </ProviderMock>
        );
        expect(form.toJSON()).toMatchSnapshot();
    });
});
