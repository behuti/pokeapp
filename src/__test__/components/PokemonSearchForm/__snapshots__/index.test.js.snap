// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Snapshot should match snapshot 1`] = `
<div
  className="sc-bdvvaa frsVvv pokemon-search"
>
  <h1
    className="pokemon-search--title"
  >
    What Pokemon are you looking for?
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="pokemon-search__input"
      id="pokemonName"
      name="pokemonName"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Pokemon Name or id"
      type="text"
      value=""
    />
    <br />
    <button
      className="sc-gsDJrp jamQIq"
      id="clearButton"
      onClick={[Function]}
    >
      Clear
    </button>
    <button
      className="sc-dkPtyc bspwyx"
      id="submitButton"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
